presto-spark-base/src/main/java/com/facebook/presto/spark/execution/property/Properties.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution.property;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A class that stores key value properties. It also provides {@link PropertiesPopulator} interfaces of populating/materializing the properties. See {@link FilePropertiesPopulator}
+ * for a concrete implementation of a {@link PropertiesPopulator}.
+ */
+public class Properties
+{
+    private PropertiesPopulator populator;
+    private ImmutableMap<String, String> propertiesMap;
+
+    public Properties(PropertiesPopulator populator, ImmutableMap<String, String> propertiesMap)
+    {
+        this.populator = populator;
+        this.propertiesMap = propertiesMap;
+    }
+
+    public Map<String, String> getProperties()
+    {
+        return propertiesMap;
+    }
+
+    public void populateProperties()
+            throws Exception
+    {
+        populator.populate(propertiesMap);
+    }
+
+    public abstract static class PropertiesBuilder<T extends PropertiesBuilder<T>>
+    {
+        protected final HashMap<String, String> propertiesMap = new HashMap<>();
+        private PropertiesPopulator populator;
+
+        public T setPopulator(PropertiesPopulator populator)
```

Comment:
- same, return `PropertiesBuilder<T>`
- can we actually remove this method and make it a constructor?

Reply:
N.A. anymore